src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -157,35 +174,41 @@ void runtime::cpu::pass::CPULayout::set_default_layouts(
     {
         const auto& output = input.get_output();
         auto tv = output.get_tensor_view();
-        auto tvt = tv->get_tensor_view_type();
-        auto rank = tvt->get_shape().size();
+        auto et = tv->get_tensor_view_type()->get_element_type();
+        auto shape = tv->get_tensor_view_type()->get_shape();
         auto tvl = tv->get_tensor_view_layout();
         auto cpu_tvl = dynamic_cast<runtime::cpu::LayoutDescriptor*>(tvl.get());
-        if (cpu_tvl && cpu_tvl->get_mkldnn_format() != memory::format::format_undef &&
-            !runtime::cpu::mkldnn_utils::compare_mkldnn_formats(
-                cpu_tvl->get_mkldnn_format(),
-                runtime::cpu::mkldnn_utils::CreateNativeDataFormat(*cpu_tvl)))
+
+        if (cpu_tvl && cpu_tvl->is_mkldnn_layout())
         {
-            auto native_axis_order =
-                ngraph::runtime::cpu::LayoutDescriptor::create_native_axis_order(rank);
-            auto layout =
-                std::make_shared<ngraph::runtime::cpu::LayoutDescriptor>(*tv, native_axis_order);
-            layout->set_mkldnn_format(runtime::cpu::mkldnn_utils::CreateNativeDataFormat(*cpu_tvl));
-            auto new_node = std::shared_ptr<Node>(
-                new runtime::cpu::op::ConvertLayout(output.get_node(), output.get_index(), layout));
-            new_args.push_back(new_node);
-            if (use_replace)
+            auto native_md =
+                mkldnn_utils::create_blocked_mkldnn_md(shape, cpu_tvl->get_strides(), et);
+            if (!mkldnn_utils::compare_mkldnn_mds(cpu_tvl->get_mkldnn_md(), native_md))
```

Comment:
little confused here, if the node already have mkldnn layout and if we again create mkldnn md based on tv->strides . isn't it both are same?. and here when we meant native_md is it NCHW? (framework format)

Reply:
cpu_tvl->get_strides() is always row-major. get_mkldnn_md can be whatever the layout pass has on that tensor.